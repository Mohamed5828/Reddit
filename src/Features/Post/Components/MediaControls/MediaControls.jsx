// Import styled components
import { MediaControlsContainer, StyledButton } from "./MediaControls.styled";

// Import icons
import { IoImageOutline } from "react-icons/io5";
import { SlSocialYoutube } from "react-icons/sl";

// Import hooks
import { useRef } from "react";
import { OverlayTrigger, Tooltip } from "react-bootstrap";

/**
 * The media control in the draft editor
 * media controls like image & video
 *
 * @param {Function} addImage - Function to handle adding image
 * @param {Function} addVideo - Function to handle adding video
 * @returns {React.Component} - The media control in the draft editor
 */
const MediaControls = ({ addImage, addVideo }) => {
  // Ref for the image input
  const imageRef = useRef();

  // Ref for the video input
  const videoRef = useRef();
  return (
    <MediaControlsContainer>
      <OverlayTrigger
        placement="top"
        overlay={<Tooltip id="button-tooltip-2">Add an image</Tooltip>}
      >
        <StyledButton
          id="Add-an-image"
          onClick={() => imageRef.current.click()}
        >
          <IoImageOutline size={25} />
        </StyledButton>
      </OverlayTrigger>
      <input
        as={"input"}
        type="file"
        accept="image/*"
        onChange={(e) => addImage(e.target.files[0])}
        ref={imageRef}
        hidden
      />
      <OverlayTrigger
        placement="top"
        overlay={<Tooltip id="button-tooltip-2">Add a video</Tooltip>}
      >
        <StyledButton id="Add-a-video" onClick={() => videoRef.current.click()}>
          <SlSocialYoutube size={25} />
        </StyledButton>
      </OverlayTrigger>
      <input
        type="file"
        accept="video/*"
        onChange={(e) => addVideo(e.target.files[0])}
        ref={videoRef}
        hidden
      />
    </MediaControlsContainer>
  );
};

export default MediaControls;
