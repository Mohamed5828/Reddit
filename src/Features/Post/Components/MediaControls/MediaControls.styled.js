// Import styled
import styled from "styled-components";

export const MediaControlsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  max-width: 100%;
`;

export const StyledButton = styled.span`
  cursor: pointer;
  padding: 2px 0;
  display: inline-block;
  color: ${({ active, theme }) =>
    active ? "black" : theme.borderColor.primary};
  border-radius: 4px;
  outline: none;
  padding: 4px;
  &:hover {
    background-color: ${({ theme }) => theme.background.muted};
  }
`;
