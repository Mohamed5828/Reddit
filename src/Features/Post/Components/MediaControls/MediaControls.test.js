import { shallow } from "enzyme";
import React from "react";

// Import components
import MediaControls from "./MediaControls";

describe("Media controls in draft editor", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MediaControls />);
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
