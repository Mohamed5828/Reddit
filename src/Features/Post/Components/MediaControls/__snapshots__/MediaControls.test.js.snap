// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media controls in draft editor should render without crashing 1`] = `
<styled.div>
  <OverlayTrigger
    defaultShow={false}
    overlay={
      <Tooltip
        id="button-tooltip-2"
        placement="right"
      >
        Add an image
      </Tooltip>
    }
    placement="top"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.span
      onClick={[Function]}
    >
      <IoImageOutline
        size={25}
      />
    </styled.span>
  </OverlayTrigger>
  <input
    accept="image/*"
    as="input"
    hidden={true}
    onChange={[Function]}
    type="file"
  />
  <OverlayTrigger
    defaultShow={false}
    overlay={
      <Tooltip
        id="button-tooltip-2"
        placement="right"
      >
        Add a video
      </Tooltip>
    }
    placement="top"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.span
      onClick={[Function]}
    >
      <SlSocialYoutube
        size={25}
      />
    </styled.span>
  </OverlayTrigger>
  <input
    accept="video/*"
    hidden={true}
    onChange={[Function]}
    type="file"
  />
</styled.div>
`;
